File: src/app/models/order.ts
// models/order.ts
export interface Order {
  OrderId?: number; 
  ClientId: number; 
  ClientName?: string; // Nombre del cliente (agregado para la vista)
  OrderDate: string; 
  Total: number; 
  Client?: { 
    Name: string; // Nombre del cliente recibido del backend
  }; 
}

File: ControlStockBack/Models/Order.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace ControlStockBack.Models
{
    public class Order
    {
        public int OrderId { get; set; }
        public int ClientId { get; set; }
        public Client Client { get; set; } = new Client();  // Inicializa como nuevo objeto si no es null.
        public DateTime OrderDate { get; set; }
        public decimal Total { get; set; }
    }

}